### ISSUE TITLE
Training become very slow after some epochs!

### ISSUE THREAD

**Issue by duyhominhnguyen** [lmotte/graph-prediction-with-fused-gromov-wasserstein#1]
Hi authors,

Thank you for your interesting work! I'm running your provided example but the training step becomes very slow after epochs 40; especially at ``` loss_i = self.loss(pred, Y[i])``` and ```pred = self.predict(X[i])``` in ```method.py```. For e.g, times used for predict at epoch 49 is 4.12 secs, compute loss is 5.66 secs while at initial epochs they are just approximately 0.0025 and 0.0026 respectively. Do you have any ideas why these issues occur? Thank you.
